### PR TITLE
feat(types): add types for sd-notify

### PR DESCRIPTION
**Pull request by Cheprer**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/sd-notify/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
+// Type definitions for sd-notify 2.8
+// Project: https://github.com/systemd/node-sd-notify#readme
+// Definitions by: Jan Beseda <https://github.com/Cheprer>
+//                 Rory Bradford <https://github.com/roryrjb>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// Minimum TypeScript Version: 4.7
```
Comment (peterblazejewicz) [inline types/sd-notify/index.d.ts:6]:
```suggestion
```
sorry, what's the language feature requiring this specific version?

File: types/sd-notify/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+ * @param interval milliseconds
+ */
+export function startWatchdogMode(interval: number): void;
+export function stopWatchdogMode(): void;
+/**
+ * @param text status string to systemd, which will append to the service's log
+ */
+export function sendStatus(text: string): void;
+export const journalPrint: JournalPrint;
+export interface Log {
```
Comment (peterblazejewicz) [inline types/sd-notify/index.d.ts:21]: should this one be also exported as `log` symbol?

File: types/sd-notify/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+ * @returns the amount of milliseconds watchdog has been set to, or 0 if it has not been set
+ */
+export function watchdogInterval(): number;
+
+/**
+ * Unwrapped sendStatus. Sends raw string without prepending `STATUS=` and appending new line char `\n`
+ */
+export function sendState(args: string[]): void;
+
+export as namespace SdNotify;
```
Comment (peterblazejewicz) [inline types/sd-notify/index.d.ts:52]:
```suggestion
```
there is no global symbol in this module, why this one is required?

File: types/sd-notify/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
+// Type definitions for sd-notify 2.8
+// Project: https://github.com/systemd/node-sd-notify#readme
+// Definitions by: Jan Beseda <https://github.com/Cheprer>
+//                 Rory Bradford <https://github.com/roryrjb>
```
Comment (peterblazejewicz) [inline types/sd-notify/index.d.ts:4]:
```suggestion
//                 Rory Bradford <https://github.com/roryrjb>
```
the ACK of this user is required, if you want to have this handle in the source.
DT is sending notifications to owners, so ACK would be required here, thanks!

Comment (roryrjb) [reply types/sd-notify/index.d.ts:4]: :+1: from me

File: types/sd-notify/index.d.ts (hunk 1 above)
Comment (Cheprer) [reply types/sd-notify/index.d.ts:6]: Nothing specific, it is what the project I'm intended to use it uses.

Comment (peterblazejewicz) [reply types/sd-notify/index.d.ts]: So there is no requirement for TS version, we promote interop between types (like not enforcing specific TS version), can this one be removed?

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@Cheprer thanks!

**Comment by Cheprer** [comment]
Ready to merge

**Pull request merged**